fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -870,6 +875,28 @@ where
     }
 }
 
+fn load_contract_code_from_storage<S>(
+    storage: &S,
+    contract_len: usize,
+    contract_id: ContractId,
+) -> Result<Vec<u8>, RuntimeError<<S as InterpreterStorage>::DataError>>
+where
+    S: InterpreterStorage,
+{
+    let mut contract_buffer: Vec<u8> = alloc::vec![0u8; contract_len];
+    storage
+        .read_contract(&contract_id, &mut contract_buffer)
+        .transpose()
+        .ok_or(PanicReason::ContractNotFound)?
+        .map_err(RuntimeError::Storage)?;
+
+    if contract_buffer.len() != contract_len {
```

Comment:
If it's already allocated to to `contracct_len` size, will the `len` call ever not be `contract_len`?

Reply:
Yes, this is a leftover from refactoring, most probably this check should not be there. 
(Btw, I noticed that we have a slightly different implementation of the same function: https://github.com/FuelLabs/fuel-vm/blob/a6abe47f14a91f7498d27ebf956a882ecb52aa44/fuel-vm/src/interpreter/flow.rs#L631, so maybe we should factor out the common code and have only one function instead)